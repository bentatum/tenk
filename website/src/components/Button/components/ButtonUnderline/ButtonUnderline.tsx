import { ButtonProps } from "@/components/Button";
import classNames from "classnames";
import { FC } from "react";

export interface ButtonUnderlineProps {
  color: ButtonProps["color"];
  disabled?: boolean;
}

export const ButtonUnderline: FC<ButtonUnderlineProps> = ({
  color,
  disabled,
}) => {
  return (
    <span
      role="presentation"
      className={classNames(
        "mx-auto",
        "h-0.5 w-11/12",
        "absolute inset-x-0 bottom-0",
        color === "neutral" && [
          "bg-black hover:bg-neutral-700 active:bg-neutral-600 disabled:bg-neutral-300",
          "dark:bg-white dark:hover:bg-neutral-300 dark:active:bg-neutral-400 dark:disabled:bg-neutral-500",
        ],
        color === "primary" && [
          disabled
            ? "bg-neutral-300"
            : "bg-gradient-to-r from-yellow-300 via-orange-400 to-pink-400",
        ],
        {
          "bg-pink-500 hover:bg-pink-600 active:bg-pink-700 disabled:bg-pink-300":
            color === "secondary",
          "bg-red-500 hover:bg-red-600 active:bg-red-700":
            color === "destructive",
        }
      )}
    />
  );
};
export default ButtonUnderline;
