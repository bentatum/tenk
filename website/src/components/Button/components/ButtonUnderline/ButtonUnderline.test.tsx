import { render, screen } from "@testing-library/react";
import ButtonUnderline from "./ButtonUnderline";

describe("ButtonUnderline", () => {
  test("size and position", () => {
    render(<ButtonUnderline color="neutral" />);
    expect(screen.getByRole("presentation")).toHaveClass(
      "mx-auto",
      "h-0.5",
      "w-11/12",
      "absolute",
      "inset-x-0",
      "bottom-0"
    );
  });

  describe("neutral", () => {
    test("color", () => {
      render(<ButtonUnderline color="neutral" />);
      expect(screen.getByRole("presentation")).toHaveClass(
        "bg-black",
        "dark:bg-white",
        "hover:bg-neutral-700",
        "dark:hover:bg-neutral-300",
        "active:bg-neutral-600",
        "dark:active:bg-neutral-400",
        "disabled:bg-neutral-300",
        "dark:disabled:bg-neutral-500"
      );
    });

    test("snapshot", () => {
      const { container } = render(<ButtonUnderline color="neutral" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("primary", () => {
    test("color", () => {
      render(<ButtonUnderline color="primary" />);
      expect(screen.getByRole("presentation")).toHaveClass(
        "bg-gradient-to-r",
        "from-yellow-300",
        "via-orange-400",
        "to-pink-400"
      );
    });

    test('disabled', () => {
      render(<ButtonUnderline color="primary" disabled />);
      expect(screen.getByRole("presentation")).toHaveClass(
        "bg-neutral-300"
      );
    })

    test("snapshot", () => {
      const { container } = render(<ButtonUnderline color="primary" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("secondary", () => {
    test("color", () => {
      render(<ButtonUnderline color="secondary" />);
      expect(screen.getByRole("presentation")).toHaveClass(
        "bg-pink-500",
        "hover:bg-pink-600",
        "active:bg-pink-700"
      );
    });
    test("snapshot", () => {
      const { container } = render(<ButtonUnderline color="secondary" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("destructive", () => {
    test("color", () => {
      render(<ButtonUnderline color="destructive" />);
      expect(screen.getByRole("presentation")).toHaveClass(
        "bg-red-500",
        "hover:bg-red-600",
        "active:bg-red-700"
      );
    });
    test("snapshot", () => {
      const { container } = render(<ButtonUnderline color="destructive" />);
      expect(container).toMatchSnapshot();
    });
  });
});
