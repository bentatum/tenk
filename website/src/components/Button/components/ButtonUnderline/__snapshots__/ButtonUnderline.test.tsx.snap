// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonUnderline destructive snapshot 1`] = `
<div>
  <span
    class="mx-auto h-0.5 w-11/12 absolute inset-x-0 bottom-0 bg-red-500 hover:bg-red-600 active:bg-red-700"
    role="presentation"
  />
</div>
`;

exports[`ButtonUnderline neutral snapshot 1`] = `
<div>
  <span
    class="mx-auto h-0.5 w-11/12 absolute inset-x-0 bottom-0 bg-black hover:bg-neutral-700 active:bg-neutral-600 disabled:bg-neutral-300 dark:bg-white dark:hover:bg-neutral-300 dark:active:bg-neutral-400 dark:disabled:bg-neutral-500"
    role="presentation"
  />
</div>
`;

exports[`ButtonUnderline primary snapshot 1`] = `
<div>
  <span
    class="mx-auto h-0.5 w-11/12 absolute inset-x-0 bottom-0 bg-gradient-to-r from-yellow-300 via-orange-400 to-pink-400"
    role="presentation"
  />
</div>
`;

exports[`ButtonUnderline secondary snapshot 1`] = `
<div>
  <span
    class="mx-auto h-0.5 w-11/12 absolute inset-x-0 bottom-0 bg-pink-500 hover:bg-pink-600 active:bg-pink-700 disabled:bg-pink-300"
    role="presentation"
  />
</div>
`;
